lib/components/scanner_settings.dart

```diff
@@ -8,32 +8,28 @@ class ScannerSettings extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return ValueListenableBuilder<TorchState>(
-      valueListenable: scanController.torchState,
-      builder: (context, value, child) {
-        const Color iconColor = Colors.white;
-
-        return Row(
-          mainAxisSize: MainAxisSize.max,
-          mainAxisAlignment: MainAxisAlignment.spaceBetween,
-          children: [
-            IconButton(
-              onPressed: () => Navigator.pop(context),
-              icon: const Icon(
-                Icons.close,
-                color: iconColor,
-              ),
-            ),
-            IconButton(
-              onPressed: () => scanController.toggleTorch(),
-              icon: Icon(
-                value == TorchState.on ? Icons.flash_off : Icons.flash_on,
-                color: iconColor,
-              ),
-            ),
-          ],
-        );
-      },
+    const Color iconColor = Colors.white;
```

Comment:
I think we still need to listen on `scanController` here, otherwise the widget tree won't pick up the new torch state.

Reply:
Okay, seems to work now.